### PR TITLE
Initial Project setup 🎉 🚀 ✔️ 

### PR DESCRIPTION
**Pull request by Nizamuddin4493**
# Initial Project setup 🎉 🚀 ✔️ 
- Implemented classes and objects in Ruby.
- Implemented encapsulation and inheritance with Ruby.
- Runed a program using the command line.
- Created class Person.
- Created class Student.
- Created class Teacher.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Person
+  attr_accessor :id, :name, :age, :parent_permission
```
Comment (OLIPLICHE) [inline person.rb]: - Please, Here `id` should only be able to be retrieved so only the Getter is defined. Kindly declare it as : `attr_reader :id`
- Please[ open here to learn more](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#useful-shortcuts-attr_reader-attr_writer-and-attr_accessor)

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  def initialize(id, age, name = 'unknown', parent_permission: true)
+    @id = id
```
Comment (OLIPLICHE) [inline person.rb]: - Please `id` is going to be a random integer between 1 and 1000:
- Open this to [see an example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars)
- For this is Optional: I suggest that you should not initialize `id`. Apply this in `teacher` and `student` classes as well

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Nizamuddin4493, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used (Pull Request is made from the feature branch into the dev branch) 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require_relative 'person'
+
+class Student < Person
+  attr_accessor :id, :classroom
+
+  def initialize(age, classroom, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission)
+    @id = Random.rand(1..1000)
+    @classroom = classroom
+  end
```
Comment (thecodechaser) [inline student.rb]: - [ ] Please remove `id` from this class. `id` is already defined as an instance variable in the parent class(person) that is also available for children's classes.

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_accessor :id, :specialization
+
+  def initialize(age, specialization, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission)
+    @id = Random.rand(1..1000)
+    @specialization = specialization
+  end
```
Comment (thecodechaser) [inline teacher.rb]: - [ ] Please remove `id` from this class. `id` is already defined as an instance variable in the parent class(person) that is also available for children's classes.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Nizamuddin4493👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi @Nizamuddin4493, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**